operator-common/src/main/java/io/strimzi/operator/common/metrics/MetricsHolder.java

```diff
@@ -5,31 +5,79 @@
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
 import java.util.function.Function;
+import java.util.stream.Collectors;
 
 /**
  * Abstract base class holding common metrics used by operators and controllers.
  * Subclasses can add more specialized metrics.
  */
 public abstract class MetricsHolder {
-    protected static final String METRICS_PREFIX = "strimzi.";
+    /**
+     * Prefix used for metrics provided by Strimzi operators
+     */
+    public static final String METRICS_PREFIX = "strimzi.";
+    /**
+     * Metric name for number of reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS = METRICS_PREFIX + "reconciliations";
+    /**
+     * Metric name for number of periodic reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS_PERIODICAL = METRICS_PREFIX + "reconciliations.periodical";
+    /**
+     * Metric name for number of failed reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS_FAILED = METRICS_PREFIX + "reconciliations.failed";
+    /**
+     * Metric name for number of successful reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS_SUCCESSFUL = METRICS_PREFIX + "reconciliations.successful";
+    /**
+     * Metric name for duration of reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS_DURATION = METRICS_PREFIX + "reconciliations.duration";
+    /**
+     * Metric name for number of locked reconciliations.
+     */
+    public static final String METRICS_RECONCILIATIONS_LOCKED = METRICS_PREFIX + "reconciliations.locked";
+    /**
+     * Metric name for number of resources managed by the operator.
+     */
+    public static final String METRICS_RESOURCES = METRICS_PREFIX + "resources";
+    /**
+     * Metric name for number of paused resources.
+     */
+    public static final String METRICS_RESOURCES_PAUSED = METRICS_PREFIX + "resources.paused";
+    /**
+     * Metric name for server certificate expiration timestamp in ms.
+     */
+    public static final String METRICS_SERVER_CERTIFICATE_EXPIRATION_MS = METRICS_PREFIX + "server.certificate.expiration.ms";
+    /**
+     * Metric name for client certificate expiration timestamp in ms.
+     */
+    public static final String METRICS_CLIENT_CERTIFICATE_EXPIRATION_MS = METRICS_PREFIX + "client.certificate.expiration.ms";
 
     protected final String kind;
     protected final Labels selectorLabels;
     protected final MetricsProvider metricsProvider;
 
     protected final Map<String, AtomicInteger> resourceCounterMap = new ConcurrentHashMap<>(1);
     protected final Map<String, AtomicInteger> pausedResourceCounterMap = new ConcurrentHashMap<>(1);
+    protected final Map<String, AtomicLong> serverCertificateExpirationMap = new ConcurrentHashMap<>(1);
+    protected final Map<String, AtomicLong> clientCertificateExpirationMap = new ConcurrentHashMap<>(1);
```

Comment:
Same as above, unless we want to use a single metric and tags, it should ba named `clusterCaCertificateExpirationMap` and `clientsCa...`.